packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -117,36 +163,53 @@ export class DataTable {
    */
   renderPromiseResolve = null;
 
+  popperInstance: Instance = null;
+
+  settingsButton: HTMLElement = null;
+
+  settingsUpdateButton: HTMLFwButtonElement = null;
+
+  settingsResetButton: HTMLElement = null;
+
+  settingsInput: HTMLFwInputElement = null;
+
+  settingsDragContainer: HTMLFwDragContainerElement = null;
+
+  settings: HTMLElement = null;
+
+  tableContainer: HTMLElement = null;
+
   /**
    * componentWillLoad lifecycle event
    */
   componentWillLoad() {
     this.columnOrdering(this.columns);
+    if (localStorage && this.autoSaveSettings) {
+      const tableId = this.el.id ? `-${this.el.id}` : '';
```

Comment:
can we set a unique ID if the `id` is not set by the user ?

Reply:
As per our discussion, added error message for when autosave is present and id is missing for the table.